rstest_macros/src/parse/rstest/files.rs

```diff
@@ -16,11 +16,54 @@ use crate::{
     utils::attr_is,
 };
 
+/// Replace environment variables in a string.
+fn replace_env_vars(attr: &LitStrAttr, ignore_missing: bool) -> Result<String, syn::Error> {
+    let re =
+        Regex::new(r"\$([a-zA-Z_][a-zA-Z0-9_]*)|\$\{([^}]*)}").expect("Could not build the regex");
+
+    let path = attr.value();
+    let haystack = path.as_str();
+    let mut result = String::with_capacity(attr.value().len());
+    let mut last_match = 0;
+
+    for caps in re.captures_iter(haystack) {
+        let m = caps.get(0).expect("The regex should have matched");
+        let var_name = caps
+            .get(1)
+            .or_else(|| caps.get(2))
+            .expect("The regex should have matched either $VAR or ${VAR}")
+            .as_str();
+
+        let replacement = match env::var(var_name) {
+            Ok(val) => val,
+            Err(_) if ignore_missing => String::new(),
+            Err(_) => {
+                return Err(attr.error(&format!(
+                    "Could not find the environment variable {:?}",
+                    var_name
+                )))
+            }
+        };
+
+        // Make sure cargo would rerun if the value of this environment variable changes.
+        println!("cargo::rerun-if-env-changed={var_name}");
```

Comment:
It works in my project. Just rerunning `cargo test` without changing the environment variable will not rebuild the project, but changing it will rebuild.